### ISSUE TITLE
CDC becomes unhealthy in a couple of seconds after startup and doesn't seem to interact with Kafka or poll the OUTBOX

### ISSUE THREAD

**Issue by amirh715** [eventuate-foundation/eventuate-cdc#127]
I'm trying to use polling with CDC and not only it becomes unhealthy just after startup but also doesn't seem to poll the `OUTBOX` and publish to Kafka.

I've tested Kafka instance itself and it's working properly. MySQL is working fine too and messages are inserted into `OUTBOX`.

**docker-compose.yml**
```
version: '3'
name: EasyBuy
networks:
  app-tier:
    name: app-tier
services:
  zookeeper:
    image: bitnami/zookeeper:latest
    networks:
      - app-tier
    ports:
      - 2181:2181
    environment:
      ALLOW_ANONYMOUS_LOGIN: yes
  kafka:
    image: bitnami/kafka
    networks:
      - app-tier
    ports:
      - 9092:9092
    depends_on:
      - zookeeper
    environment:
      KAFKA_BROKER_ID: 1
      KAFKA_CFG_LISTENERS: PLAINTEXT://:9092
      KAFKA_CFG_ADVERTISED_LISTENERS: PLAINTEXT://kafka:9092
      KAFKA_CFG_ZOOKEEPER_CONNECT: zookeeper:2181
      ALLOW_PLAINTEXT_LISTENER: yes
  eventuate-cdc:
    image: eventuateio/eventuate-cdc-service
    networks:
      - app-tier
    ports:
      - 8099:8080
    depends_on:
      - mysql
      - zookeeper
      - kafka
    environment:
      EVENTUATELOCAL_KAFKA_BOOTSTRAP_SERVERS: PLAINTEXT://kafka:9092
      EVENTUATELOCAL_ZOOKEEPER_CONNECTION_STRING: zookeeper:2181
      EVENTUATE_CDC_READER_READER1_TYPE: polling
      EVENTUATE_CDC_READER_READER1_DATASOURCEURL: jdbc:mysql://mysql:3306/eventuate?allowPublicKeyRetrieval=true&useSSL=false
      EVENTUATE_CDC_READER_READER1_DATASOURCEUSERNAME: root
      EVENTUATE_CDC_READER_READER1_DATASOURCEPASSWORD: password
      EVENTUATE_CDC_READER_READER1_DATASOURCEDRIVERCLASSNAME: com.mysql.jdbc.Driver
      EVENTUATE_CDC_READER_READER1_LEADERSHIPLOCKPATH: /eventuatelocal/cdc/leader/eventuatelocal
      EVENTUATE_CDC_PIPELINE_PIPELINE1_TYPE: eventuate-tram
      EVENTUATE_CDC_PIPELINE_PIPELINE1_READER: reader1
      EVENTUATE_CDC_PIPELINE_PIEPLINE1_DATASOURCEDRIVERCLASSNAME: com.mysql.jdbc.Driver
      EVENTUATE_CDC_SERVICE_DRY_RUN: true
      USE_DB_ID: true
  mysql:
    image: mysql
    networks:
      - app-tier
    ports:
      - 3308:3306
    environment:
      MYSQL_ROOT_PASSWORD: password
      MYSQL_DATABASE: eventuate
  oneaccess:
    image: 2350866ca61c
    networks:
      - app-tier
    depends_on:
      - eventuate-cdc
      - mysql
    ports:
      - 8081:8081

```
**docker container inspect output**
```
[
    {
        "Id": "50499fad0be13c0492ed36453e9e11f2fbf8cfe61618d83f8fabf4035dab4948",
        "Created": "2022-07-24T12:42:13.911479919Z",
        "Path": "/bin/sh",
        "Args": [
            "-c",
            "java ${JAVA_OPTS} -jar eventuate-cdc-service-*.jar ${CDC_OPTS}"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 16144,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2022-07-24T12:42:15.257853719Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "unhealthy",
                "FailingStreak": 17,
                "Log": [
                    {
                        "Start": "2022-07-24T12:46:56.340714535Z",
                        "End": "2022-07-24T12:46:56.415488475Z",
                        "ExitCode": 1,
                        "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 8080: Connection refused\n"
                    },
                    {
                        "Start": "2022-07-24T12:47:16.419957209Z",
                        "End": "2022-07-24T12:47:16.489136769Z",
                        "ExitCode": 1,
                        "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 8080: Connection refused\n"
                    },
                    {
                        "Start": "2022-07-24T12:47:36.496356464Z",
                        "End": "2022-07-24T12:47:36.567784509Z",
                        "ExitCode": 1,
                        "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 8080: Connection refused\n"
                    },
                    {
                        "Start": "2022-07-24T12:47:56.579013074Z",
                        "End": "2022-07-24T12:47:56.656907472Z",
                        "ExitCode": 1,
                        "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 8080: Connection refused\n"
                    },
                    {
                        "Start": "2022-07-24T12:48:16.667768271Z",
                        "End": "2022-07-24T12:48:16.73345765Z",
                        "ExitCode": 1,
                        "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 8080: Connection refused\n"
                    }
                ]
            }
        },
        "Image": "sha256:4c549b0c4475ff9711448c5cecaf8cc44308956314c3c33032a04bd21b765c58",
        "ResolvConfPath": "/var/lib/docker/containers/50499fad0be13c0492ed36453e9e11f2fbf8cfe61618d83f8fabf4035dab4948/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/50499fad0be13c0492ed36453e9e11f2fbf8cfe61618d83f8fabf4035dab4948/hostname",
        "HostsPath": "/var/lib/docker/containers/50499fad0be13c0492ed36453e9e11f2fbf8cfe61618d83f8fabf4035dab4948/hosts",
        "LogPath": "/var/lib/docker/containers/50499fad0be13c0492ed36453e9e11f2fbf8cfe61618d83f8fabf4035dab4948/50499fad0be13c0492ed36453e9e11f2fbf8cfe61618d83f8fabf4035dab4948-json.log",
        "Name": "/easybuy-eventuate-cdc-1",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "app-tier",
            "PortBindings": {
                "8080/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "8099"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "private",
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": null,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/a3aa05dbbd07bcc1333271f71135972b863a9b10d3b84927b38faa6b0b32b87f-init/diff:/var/lib/docker/overlay2/9d7997166944e02db540cbb176fb4e0a125b748ed1545b2097133567060daee7/diff:/var/lib/docker/overlay2/4d5bad9f17b10aaf379038c7d83e790559977e34697e871b93445b394d8fae34/diff:/var/lib/docker/overlay2/6c5357ef4ce149cb81fb87e731a463f55c8867920f1a2d5577d4b5453b8a306d/diff:/var/lib/docker/overlay2/dceb283762265ac2eb40af0f49216358d2b46366d3a3215f1586c1a6193a9606/diff:/var/lib/docker/overlay2/7d902be7b192e36258fa3c6975eb41bb4f00a768c18aee72b2b7266ece04b0fa/diff:/var/lib/docker/overlay2/5867946678e989a868743185364b1ccbd2a8bda9d584b42b2d47cbcc11358e6d/diff",
                "MergedDir": "/var/lib/docker/overlay2/a3aa05dbbd07bcc1333271f71135972b863a9b10d3b84927b38faa6b0b32b87f/merged",
                "UpperDir": "/var/lib/docker/overlay2/a3aa05dbbd07bcc1333271f71135972b863a9b10d3b84927b38faa6b0b32b87f/diff",
                "WorkDir": "/var/lib/docker/overlay2/a3aa05dbbd07bcc1333271f71135972b863a9b10d3b84927b38faa6b0b32b87f/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [],
        "Config": {
            "Hostname": "50499fad0be1",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": true,
            "AttachStderr": true,
            "ExposedPorts": {
                "8080/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "EVENTUATE_CDC_READER_READER1_TYPE=polling",
                "EVENTUATE_CDC_READER_READER1_DATASOURCEUSERNAME=root",
                "EVENTUATE_CDC_READER_READER1_DATASOURCEURL=jdbc:mysql://mysql:3306/eventuate?allowPublicKeyRetrieval=true\u0026useSSL=false",
                "EVENTUATE_CDC_READER_READER1_LEADERSHIPLOCKPATH=/eventuatelocal/cdc/leader/eventuatelocal",
                "EVENTUATELOCAL_KAFKA_BOOTSTRAP_SERVERS=PLAINTEXT://kafka:9092",
                "EVENTUATE_CDC_SERVICE_DRY_RUN=true",
                "EVENTUATE_CDC_PIPELINE_PIEPLINE1_DATASOURCEDRIVERCLASSNAME=com.mysql.jdbc.Driver",
                "EVENTUATE_CDC_READER_READER1_DATASOURCEPASSWORD=password",
                "EVENTUATELOCAL_ZOOKEEPER_CONNECTION_STRING=zookeeper:2181",
                "EVENTUATE_CDC_READER_READER1_DATASOURCEDRIVERCLASSNAME=com.mysql.jdbc.Driver",
                "EVENTUATE_CDC_PIPELINE_PIPELINE1_READER=reader1",
                "EVENTUATE_CDC_PIPELINE_PIPELINE1_TYPE=eventuate-tram",
                "USE_DB_ID=true",
                "PATH=/usr/local/openjdk-8/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "LANG=C.UTF-8",
                "JAVA_HOME=/usr/local/openjdk-8",
                "JAVA_VERSION=8u252",
                "JAVA_BASE_URL=https://github.com/AdoptOpenJDK/openjdk8-upstream-binaries/releases/download/jdk8u252-b09/OpenJDK8U-jdk_",
                "JAVA_URL_VERSION=8u252b09"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "java ${JAVA_OPTS} -jar eventuate-cdc-service-*.jar ${CDC_OPTS}"
            ],
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "curl -f http://localhost:8080/actuator/health || exit 1"
                ],
                "Interval": 20000000000,
                "StartPeriod": 30000000000
            },
            "Image": "eventuateio/eventuate-cdc-service",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "com.docker.compose.config-hash": "d9123cc8f8944813a3cc073dcf9dc813b097df8dbee32885a9bc642e049737a2",
                "com.docker.compose.container-number": "1",
                "com.docker.compose.depends_on": "mysql:service_started,zookeeper:service_started,kafka:service_started",
                "com.docker.compose.image": "sha256:4c549b0c4475ff9711448c5cecaf8cc44308956314c3c33032a04bd21b765c58",
                "com.docker.compose.oneoff": "False",
                "com.docker.compose.project": "easybuy",
                "com.docker.compose.project.config_files": "/home/dada/Desktop/Docker files/docker-compose.yml",
                "com.docker.compose.project.working_dir": "/home/dada/Desktop/Docker files",
                "com.docker.compose.service": "eventuate-cdc",
                "com.docker.compose.version": "2.5.0"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "0c53006d74f51af54963e8cef9389eec5c9837ce574fcef8eaf36ec8fee18c87",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "8080/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "8099"
                    }
                ]
            },
            "SandboxKey": "/var/run/docker/netns/0c53006d74f5",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "app-tier": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "easybuy-eventuate-cdc-1",
                        "eventuate-cdc",
                        "50499fad0be1"
                    ],
                    "NetworkID": "fd31e210af9d11866c3817c32b1cd2fb50ecac385a0599d5078a8bc4d2c4a533",
                    "EndpointID": "9bb64821c3f005fd43af5f1495b68a8ba47a3eb124b02e4d37d50b3cbf82a963",
                    "Gateway": "172.21.0.1",
                    "IPAddress": "172.21.0.5",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:15:00:05",
                    "DriverOpts": null
                }
            }
        }
    }
]

```

**Comment by cer** [comment]
Please provide the output of the following:

* `docker logs <cdc-container>`
* `curl http://localhost:8099/actuator/health`
*

**Comment by amirh715** [comment]
**docker logs <cdc-container>**

```
#######
#        #    #  ######  #    #   #####  #    #    ##     #####  ######
#        #    #  #       ##   #     #    #    #   #  #      #    #
#####    #    #  #####   # #  #     #    #    #  #    #     #    #####
#        #    #  #       #  # #     #    #    #  ######     #    #
#         #  #   #       #   ##     #    #    #  #    #     #    #
#######    ##    ######  #    #     #     ####   #    #     #    ######

2022-07-24 17:33:28.347  INFO 9 --- [           main] i.e.cdc.main.EventuateCdcServiceMain     : Starting EventuateCdcServiceMain on 7278a2ae24c5 with PID 9 (/eventuate-cdc-service-0.10.0-SNAPSHOT.jar started by root in /)
2022-07-24 17:33:28.388  INFO 9 --- [           main] i.e.cdc.main.EventuateCdcServiceMain     : No active profile set, falling back to default profiles: default
2022-07-24 17:33:36.550  INFO 9 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2022-07-24 17:33:36.646  INFO 9 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2022-07-24 17:33:36.648  INFO 9 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet Engine: Apache Tomcat/9.0.13
2022-07-24 17:33:36.679  INFO 9 --- [           main] o.a.catalina.core.AprLifecycleListener   : The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: [/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib]
2022-07-24 17:33:36.908  INFO 9 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2022-07-24 17:33:36.910  INFO 9 --- [           main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 8146 ms
2022-07-24 17:33:38.821  INFO 9 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2022-07-24 17:33:38.982  WARN 9 --- [           main] o.a.c.retry.ExponentialBackoffRetry      : maxRetries too large (2147483647). Pinning to 29
2022-07-24 17:33:39.103  INFO 9 --- [           main] o.a.c.f.imps.CuratorFrameworkImpl        : Starting
2022-07-24 17:33:39.118  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:zookeeper.version=3.4.6-1569965, built on 02/20/2014 09:09 GMT
2022-07-24 17:33:39.119  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:host.name=7278a2ae24c5
2022-07-24 17:33:39.119  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:java.version=1.8.0_252
2022-07-24 17:33:39.119  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:java.vendor=Oracle Corporation
2022-07-24 17:33:39.120  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:java.home=/usr/local/openjdk-8/jre
2022-07-24 17:33:39.120  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:java.class.path=eventuate-cdc-service-0.10.0-SNAPSHOT.jar
2022-07-24 17:33:39.120  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:java.library.path=/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
2022-07-24 17:33:39.120  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:java.io.tmpdir=/tmp
2022-07-24 17:33:39.121  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:java.compiler=<NA>
2022-07-24 17:33:39.123  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:os.name=Linux
2022-07-24 17:33:39.124  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:os.arch=amd64
2022-07-24 17:33:39.125  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:os.version=5.10.104-linuxkit
2022-07-24 17:33:39.126  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:user.name=root
2022-07-24 17:33:39.126  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:user.home=/root
2022-07-24 17:33:39.126  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Client environment:user.dir=/
2022-07-24 17:33:39.127  INFO 9 --- [           main] org.apache.zookeeper.ZooKeeper           : Initiating client connection, connectString=zookeeper:2181 sessionTimeout=60000 watcher=org.apache.curator.ConnectionState@5c86dbc5
2022-07-24 17:33:39.181  INFO 9 --- [.app-tier:2181)] org.apache.zookeeper.ClientCnxn          : Opening socket connection to server easybuy-zookeeper-1.app-tier/172.18.0.3:2181. Will not attempt to authenticate using SASL (unknown error)
2022-07-24 17:33:39.208  INFO 9 --- [.app-tier:2181)] org.apache.zookeeper.ClientCnxn          : Socket connection established to easybuy-zookeeper-1.app-tier/172.18.0.3:2181, initiating session
2022-07-24 17:33:39.229  INFO 9 --- [.app-tier:2181)] org.apache.zookeeper.ClientCnxn          : Session establishment complete on server easybuy-zookeeper-1.app-tier/172.18.0.3:2181, sessionid = 0x1000000653a0001, negotiated timeout = 40000
2022-07-24 17:33:39.242  INFO 9 --- [ain-EventThread] o.a.c.f.state.ConnectionStateManager     : State change: CONNECTED
2022-07-24 17:33:39.326  INFO 9 --- [           main] o.a.k.clients.producer.ProducerConfig    : ProducerConfig values: 
	acks = all
	batch.size = 16384
	bootstrap.servers = [PLAINTEXT://kafka:9092]
	buffer.memory = 33554432
	client.dns.lookup = default
	client.id = 
	compression.type = none
	connections.max.idle.ms = 540000
	delivery.timeout.ms = 120000
	enable.idempotence = false
	interceptor.classes = []
	key.serializer = class org.apache.kafka.common.serialization.StringSerializer
	linger.ms = 1
	max.block.ms = 60000
	max.in.flight.requests.per.connection = 5
	max.request.size = 1048576
	metadata.max.age.ms = 300000
	metric.reporters = []
	metrics.num.samples = 2
	metrics.recording.level = INFO
	metrics.sample.window.ms = 30000
	partitioner.class = class org.apache.kafka.clients.producer.internals.DefaultPartitioner
	receive.buffer.bytes = 32768
	reconnect.backoff.max.ms = 1000
	reconnect.backoff.ms = 50
	request.timeout.ms = 30000
	retries = 0
	retry.backoff.ms = 100
	sasl.client.callback.handler.class = null
	sasl.jaas.config = null
	sasl.kerberos.kinit.cmd = /usr/bin/kinit
	sasl.kerberos.min.time.before.relogin = 60000
	sasl.kerberos.service.name = null
	sasl.kerberos.ticket.renew.jitter = 0.05
	sasl.kerberos.ticket.renew.window.factor = 0.8
	sasl.login.callback.handler.class = null
	sasl.login.class = null
	sasl.login.refresh.buffer.seconds = 300
	sasl.login.refresh.min.period.seconds = 60
	sasl.login.refresh.window.factor = 0.8
	sasl.login.refresh.window.jitter = 0.05
	sasl.mechanism = GSSAPI
	security.protocol = PLAINTEXT
	send.buffer.bytes = 131072
	ssl.cipher.suites = null
	ssl.enabled.protocols = [TLSv1.2, TLSv1.1, TLSv1]
	ssl.endpoint.identification.algorithm = https
	ssl.key.password = null
	ssl.keymanager.algorithm = SunX509
	ssl.keystore.location = null
	ssl.keystore.password = null
	ssl.keystore.type = JKS
	ssl.protocol = TLS
	ssl.provider = null
	ssl.secure.random.implementation = null
	ssl.trustmanager.algorithm = PKIX
	ssl.truststore.location = null
	ssl.truststore.password = null
	ssl.truststore.type = JKS
	transaction.timeout.ms = 60000
	transactional.id = null
	value.serializer = class org.apache.kafka.common.serialization.ByteArraySerializer

2022-07-24 17:33:39.507  INFO 9 --- [           main] o.a.kafka.common.utils.AppInfoParser     : Kafka version: 2.3.0
2022-07-24 17:33:39.509  INFO 9 --- [           main] o.a.kafka.common.utils.AppInfoParser     : Kafka commitId: fc1aaa116b661c8a
2022-07-24 17:33:39.509  INFO 9 --- [           main] o.a.kafka.common.utils.AppInfoParser     : Kafka startTimeMs: 1658684019502
2022-07-24 17:33:39.613  INFO 9 --- [           main] i.e.l.u.c.p.CdcPipelineConfigurator      : Starting unified cdc pipelines
2022-07-24 17:33:39.923  INFO 9 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
2022-07-24 17:33:39.977  INFO 9 --- [ad | producer-1] org.apache.kafka.clients.Metadata        : [Producer clientId=producer-1] Cluster ID: 5CukCz0mQOaieQSdIfx-aQ

```

**Comment by amirh715** [comment]
I can't get it to work and logs don't seem to indicate any problems.

**Comment by amirh715** [comment]
@cer I've provided the logs. The health endpoint sends empty response and the logs don't indicate any problems. I've also set `EVENTUATE_CDC_PIPELINE_<PIPELINE_NAME>_EVENTUATEDATABASESCHEMA` but made no difference.

**Comment by cer** [comment]
> 2022-07-24 17:33:39.923  INFO 9 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...

It appears to be hanging while creating the DB connection pool.

Normally you will see this:

```
2022-07-26 19:52:26.171  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
....
2022-07-26 19:52:27.381  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
```

**Comment by amirh715** [comment]
> > 2022-07-24 17:33:39.923  INFO 9 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
> 
> It appears to be hanging while creating the DB connection pool.
> 
> Normally you will see this:
> 
> ```
> 2022-07-26 19:52:26.171  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Starting...
> ....
> 2022-07-26 19:52:27.381  INFO 1 --- [           main] com.zaxxer.hikari.HikariDataSource       : HikariPool-1 - Start completed.
> ```

How can I fix it?

**Comment by cer** [comment]
> How can I fix it?

Not sure. But this narrows down the problem.

One thing that would help is to specify versions/tags for all of the images in the docker-compose.yml - it's unclear what versions of the CDC and MySQL you are running.

**Comment by cer** [comment]
Also, `docker-compose version` and `docker version`.

**Comment by cer** [comment]
Specifically, please specify: `eventuateio/eventuate-cdc-service:0.13.0.RELEASE` - previously it was pulling an older version for some reason.

**Comment by cer** [comment]
## This "works" for me:

```
  eventuate-cdc:
    image: eventuateio/eventuate-cdc-service:0.13.0.RELEASE
    networks:
      - app-tier
    ports:
      - 8099:8080
    depends_on:
      - mysql
      - zookeeper
      - kafka
    environment:
      EVENTUATELOCAL_KAFKA_BOOTSTRAP_SERVERS: PLAINTEXT://kafka:9092
      EVENTUATELOCAL_ZOOKEEPER_CONNECTION_STRING: zookeeper:2181
      EVENTUATE_CDC_READER_READER1_TYPE: polling
      EVENTUATE_CDC_READER_READER1_DATASOURCEURL: jdbc:mysql://mysql:3306/eventuate?allowPublicKeyRetrieval=true&useSSL=false
      EVENTUATE_CDC_READER_READER1_DATASOURCEUSERNAME: root
      EVENTUATE_CDC_READER_READER1_DATASOURCEPASSWORD: password
      EVENTUATE_CDC_READER_READER1_DATASOURCEDRIVERCLASSNAME: com.mysql.jdbc.Driver
      EVENTUATE_CDC_READER_READER1_LEADERSHIPLOCKPATH: /eventuatelocal/cdc/leader/eventuatelocal
      EVENTUATE_CDC_READER_READER1_OUTBOXID: 1

      EVENTUATE_CDC_PIPELINE_PIPELINE1_TYPE: eventuate-tram
      EVENTUATE_CDC_PIPELINE_PIPELINE1_READER: reader1
      EVENTUATE_CDC_SERVICE_DRY_RUN: "false"
      USE_DB_ID: "true"
```

## What changed

`eventuateio/eventuate-cdc-service:0.13.0.RELEASE`

* Specify image version

` EVENTUATE_CDC_PIPELINE_PIEPLINE1_DATASOURCEDRIVERCLASSNAME: com.mysql.jdbc.Driver`

* typo
* non-existent property
* needs to be removed

` EVENTUATE_CDC_READER_READER1_OUTBOXID: 1`

*  Missing property needed for DB generated `USE_DB_ID: "true"`

## What you need to do:

* Use a MySQL image with the Eventuate schema configured similar to https://github.com/eventuate-foundation/eventuate-common/tree/master/mysql

**Comment by amirh715** [comment]
OK. What I did was to use MySQL 5.7 because I figured out [this image uses it](https://hub.docker.com/layers/eventuate-mysql/eventuateio/eventuate-mysql/latest/images/sha256-31c47e8a85e152350803532aa059d48b36b60715c0adf3040298a43ba3e7a5a8?context=explore).

What is interesting is that prior to this version change no errors regarding non-existent properties or `outbox should not be null` was given (and I wonder why). Also prior to MySQL version change the health endpoint wasn't working either (but now it does).

Anyways now the main issue is resolved and CDC is working as expected.
**Thanks for being responsive.** @cer
